octodns/provider/constellix.py

```diff
@@ -82,95 +128,226 @@ def _request(self, method, path, params=None, data=None):
+        :return:
+            Response: The response from the API service
+        """
+        return self._request(
+            'POST',
+            '/domains', data={'names': [name]}
+        ).json()
+
+    @staticmethod
+    def _absolutize_value(value, zone_name):
+        """
+        Converts the fully-qualified domain name (FQDN) into its absolute
+        form by concatenating the sub-domain and the domain. Note that
+        the `zone_name` is required to have a trailing dot (otherwise the
+        zone object cannot be created).
+        :param value: string - The sub-domain name
+        :param zone_name: string - The domain name (required dot at end)
+        :return:
+            string: The absolute version of the FQDN
+        """
+        if value == '' or value == '.':
+            fqdn = zone_name
         elif not value.endswith('.'):
-            value = '{}.{}'.format(value, zone_name)
+            fqdn = '{}.{}'.format(value, zone_name)
+        else:
+            fqdn = '{}.{}'.format(value[:-1], zone_name)
 
-        return value
+        return fqdn
 
     def records(self, zone_name):
-        zone_id = self.domains.get(zone_name, False)
-        path = '/{}/records'.format(zone_id)
-
-        resp = self._request('GET', path).json()
-        for record in resp:
+        """
+        Retrieves a list of the records attached to the domain
+        :param zone_name: string - The domain name
+        :return:
+            list(dict) - A list of dictionaries where each dictionary
+            represents a record.
+        """
+        zone_id = self.domains().get(zone_name, False)
+        path = '/domains/{}/records'.format(zone_id)
+
+        record_list = self._request('GET', path).json()
+        for record in record_list:
             # change ANAME records to ALIAS
             if record['type'] == 'ANAME':
                 record['type'] = 'ALIAS'
 
             # change relative values to absolute
             value = record['value']
             if record['type'] in ['ALIAS', 'CNAME', 'MX', 'NS', 'SRV']:
-                if isinstance(value, unicode):
-                    record['value'] = self._absolutize_value(value,
-                                                             zone_name)
-                if isinstance(value, list):
+                # find out why "unicode" is producing and error
+                if isinstance(value, (str, unicode)):
+                    record['value'] = self._absolutize_value(
+                        value, zone_name
+                    )
+                else:
                     for v in value:
-                        v['value'] = self._absolutize_value(v['value'],
-                                                            zone_name)
+                        v['value'] = self._absolutize_value(
+                            v['value'], zone_name
+                        )
 
             # compress IPv6 addresses
-            if record['type'] == 'AAAA':
```

Comment:
Probably still going to need this to keep compatibility between providers. Constellix seems to return the `0` between the colons in an IPv6 address when others keep it compressed

```
* constellix (ConstellixProvider)
*   Update
*     <AaaaRecord AAAA 3600, testyzguy.io., ['2001:470:abcd:53:0:0:0:2']> ->
*     <AaaaRecord AAAA 3600, test.yzguy.io., ['2001:470:abcd:53::2']> (config)
```